gnocchi/rest/aggregates/processor.py

```diff
@@ -175,11 +179,25 @@ def aggregated(refs_and_timeseries, operations, from_timestamp=None,
             else:
                 v = values[i]
                 t = times
-            result[ident]["timestamps"].extend(t)
-            result[ident]['granularity'].extend([granularity] * len(t))
-            result[ident]['values'].extend(v)
-
-    return dict(((ident, list(six.moves.zip(result[ident]['timestamps'],
-                                            result[ident]['granularity'],
-                                            result[ident]['values'])))
-                 for ident in result))
+            result[i]["timestamps"].extend(t)
+            result[i]['granularity'].extend([granularity] * len(t))
+            result[i]['values'].extend(v)
+
+    # NOTE(sileht): the output list is ordered like the input metric list,
+    # this makes the output always ordered in the same maner.
+    output = []
+    for i, ident in enumerate(idents):
```

Comment:
this seems like it's only handling the last sampling key. i'd much prefer we not have multiple loops and this be moved with previous

`for i, ident in enumerate(idents):`

Reply:
This is something ugly I wrote... but idents are always the same whatever the sampling is.

I will refactor the code a bit to remove this oddity. Maybe into a separate patch if the change is too big.